home/models.py

```diff
@@ -1,3 +1,32 @@
-# from django.db import models
+from django.conf import settings
+from django.db import models
 
-# Create your models here.
+
+class Gender(models.TextChoices):
+    MALE = 'M', 'Male'
+    FEMALE = 'F', 'Female'
+    UNSPECIFIED = 'U', 'Unspecified'
+
+
+class Profile(models.Model):
+    '''
+    Profile will be used as the base user in the Shlifim website.
+    Profile is an extention to the imported 'User' from 'django.contrib.auth.models'.
+    Imported Fields:
+    username - Required. 150 characters or fewer. Usernames may contain alphanumeric, _, @, +, . and - characters.
+    password - Required. A hash of, and metadata about, the password. (Django doesn’t store the raw password).
+               Raw passwords can be arbitrarily long and can contain any character.
+    date_joined - A datetime designating when the account was created. Is set to the current date/time by default
+        when the account is created.
+    email - Email address.
+    is_superuser - Boolean. Designates that this user has all permissions without explicitly assigning them.
+    Added Fields:
+    gender - Default gender is set to 'Unspecified', A user can choose to specifie its gender to Male/Female.
+    is blocked - Boolean. Designates that this user won't be able to login until an admin unblockes him.
+    '''
+    user = models.OneToOneField(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, primary_key=True)
+    gender = models.CharField(max_length=1, choices=Gender.choices, default=Gender.UNSPECIFIED, blank=True, null=True)
```

Comment:
If you have a default value, why do you need to allow it to be blank or null?

Reply:
I used the exact same definition as the one in the presentation we were shown on Sunday.
Is it wrong?

Reply:
IMO yes. Blank and Null almost never make sense being both true, and blank is probably not even relevant for CharField (AFAIK a single char cannot be a blank string).

Blank means the field can contain a blank string - it only makes sens for strings and in many cases you don't want to allow it, unless the field is for something like an optional description text.

NULL means the field does not contain a value at all (Like a NULL pointer). Sometimes NULLs in the database are used to indicate values that were not specified by the used (As opposed to blank which can mean that the value was specified but was an empty string), but in this case you already designated an UNSPECIFIED value, so also allowing NULL as well does not man much sense.

If you do want to allow for NULLs, you will have to make sure you check for it everywhere in order to avoid null pointer de-referencing, so in general its better not to allow it...

Reply:
I'll change it now.
I think this change will collide with how the Profile is created in data_test_0002.
 `Profile(user=user, gender=gender).save()`
so I'll change it to `if gender ... else`

Reply:
See comment by @simzacks , the "if" statement is probably not needed there.

Reply:
Without the if it made an error, this is why I added the if else.